### PR TITLE
Adds an emag mode to the Orion Trail game

### PR DESCRIPTION
**Pull request by ghost**
By emagging an Orion Trail machine, it unlocks Realism Mode. In a nutshell, what happens in the game also has real effects on the environment and user. 

There are considerable risks, which results in a considerable reward. By winning the Realism Edition, you get a model spaceship. This seemingly innocent spaceship actually has a functional and miniature fusion reactor aboard. By flipping a switch on the bottom, it will explode after five seconds with the force of a minibomb. Logging and admin notifications exist for both winning and priming this ship.

If you lose in the game, you, of course, die yourself. Circumstances differ depending on cause(s) of death; if you starve, you instantly become malnourished in reality. If you fly into a star, you are set on fire. Flying into a black hole has no special effects other than instant death (gibbing seems a little too much, but I suppose I could make it happen).
#### Possible Events

**Raiders:** The user will hear the sounds of combat and begin to hallucinate. Alternatively, they will take a high amount of brute damage and feel a sensation of being struck.
**Disease:** The user will suffer from one of three effects:
1. A slight sense of nausea (no effect)
2. A heavy sense of nausea (brief stun)
3. An overwhelming wave of nausea (stun and vomiting)
**Turbulent Area:** The user will either be swept off their feet, causing them to take damage, or nothing will happen.
**Hull Damage:** A 3x3 area of flooring around the machine has a high probability of turning into space turf. If there is hull plating in the game, it will be repaired after a moment with plating.
**Malfunction:** The in-game food and fuel are randomized from minimum (10) to their respective maximums (80 and 60). These numbers are then divided by a random number between 1 and 2. If there is spare parts for fixing the malfunction, the food and fuel will be reset to their old amounts and chime with a different message depending on if the new amount was higher or lower than before.

I highly encourage feedback and suggestions for this.

Here's the sprite for the ship (modified from Mechoid's drone sprites)
ship

### CODE REVIEW

**Comment by xxalpha** [comment]
But what happens if you win?

**Comment by Super-neji** [comment]
Beyond cool

**Comment by Xhuis** [comment]
Alpha: It's detailed in the OP, but I worded it badly. I'll change that now

**Comment by RandomMarine** [comment]
Disease: The user will suffer from one of three effects:
1. A slight sense of nausea (no effect)
2. A heavy sense of nausea (brief stun)
3. An overwhelming wave of nausea (stun and vomiting)

How about actually give the user a virus?

**Comment by Xhuis** [comment]
I decided against that because they can do it on purpose and cause grief by injecting random people with it.

**Comment by Anonus** [comment]
This should be more deadly in general, really, and less randomly survivable.
You can survive malnutrition and being set on fire but the other stuff can just kill you.

**Comment by Xhuis** [comment]
Losing at all in the game kills you. It just makes you malnourished/set on fire _in addition_ to dying in the first place. Events are meant to be survivable but hindering, and sometimes hard to live through (like the hull breach).

I'll see about making them deadlier though. ;)

**Comment by RandomMarine** [comment]
>I decided against that because they can do it on purpose and cause grief by injecting random people with it.

Two points.

1. Almost everything in the game can be used to grief random players. There are far, far, worse and more easily done things to than playing an arcade machine modified by a syndicate item and getting people sick with the flu.

2. Non-antag virologists needs more relevance, damn it. Give them work to do!

**Comment by Xhuis** [comment]
Fair enough. I was just afraid that sickness would be treated like this:
:mask: :gun: :man:
Anyway, I'll have it give them minor diseases. Nothing major though.

**Comment by Chocobro** [comment]
this is brilliant

**Comment by Pennwick** [comment]
I think not giving actual viruses is probably  for the best. As much as just giving a cold or the flu seems pretty harmless all it takes is a ground up sheet of plasma and a dropper to create multiple round ending viri that Viroligy can't really contain. Its a really easy way to force a shuttle call.

**Comment by Xhuis** [comment]
Since there seems to be a divide on the subject of diseases, I made a poll. Please vote here
http://strawpoll.me/3823134

**Comment by PKPenguin321** [comment]
how were you handling it giving them the virus? was it like a direct proc or something? because you could have the machine just inject you with 1u of diseased blood and i don't see how that'd cause runtimes, considering there are things that automatically add reagents to you already

**Comment by Xhuis** [comment]
I had a variable (let's call it D) that had the datum for the flu in it and then I would use the AddDisease(D) proc.

**Comment by MrStonedOne** [comment]
>Hull Damage: A 3x3 area of flooring around the machine has a high probability of turning into space turf. If there is hull plating in the game, it will be repaired after a moment with plating.

There is a orion trial machine on the arrivals shuttle, either people would be able to remove floor tiles from that, griffing arrivals, something most servers don't even allow antags to do, or if this didn't do that to arrivals, then it would make emagging arrival's OT arcade better than emagging any other.

Might want to figure a way around that.

**Comment by MrStonedOne** [comment]
>Flying into a black hole has no special effects other than instant death (gibbing seems a little too much, but I suppose I could make it happen).

Make it spawn an academy singulo at their location for 5 seconds. Academy singulos can't grow or move and just eat and eat and eat while always staying at 1x1 size.

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -427,6 +445,76 @@
 			if(prob(75))
 				event = pickweight(events)
 				event()
+		if(emagged)
+			var/mob/living/carbon/M = usr //for some vars
+			if(event == "Raiders")
```
Comment (RemieRichards) [inline code/game/machinery/computer/arcade.dm]: Replace this
```
if(event == X)
if(event == Y)
```
With:
```
switch(event)
if(X)
if(Y)
```

**Comment by Xhuis** [comment]
Working on the other stuff, don't worry

**Comment by RemieRichards** [comment]
@Xhuis typically switch statements don't use ```else if```, I'm actually pretty sure it won't work with ```else if```, it just takes ```if(X)``` ```if(Y)```

**Comment by Xhuis** [comment]
All right, I'll fix that in the commit I'm testing right now

**Comment by Xhuis** [comment]
Squashed commits, fixed the switch, and made the black hole gib you.

As for the arrivals shuttle thing, I tested it and the turf doesn't get changed at all. Because it's /turf/simulated/shuttle instead of /turf/simulated/floor, there is no change. There really isn't a workaround for this, because people could theoretically just construct these machines on shuttle tiles anyway and nothing could happen.

**Comment by Jordie0608** [comment]
Two issues found during testing; After I got the singularity death I spawned myself a new mob and was killed immediately upon using the machine again as the gameover event hadn't passed since I had been killed by the singularity before getting to that screen.

I then found if the machine was in emagged gameover mode it would kill anyone who clicked on it, regardless if they were the user or not, in the 30 ticks before it un-emags itself; I did it with quick-switching mobs, but I can imagine it happening to players as well.

Presumably this means that all the emagged events can affect anyone who interacts with the machine during the time before it moves to the next day.

**Comment by Xhuis** [comment]
All right, when i get some time I'll see what I can do.

**Comment by Xhuis** [comment]
Fixed those issues although it's important to note you can still throw yourself into the singularity to kill yourself (that's pretty much a given though).

I was thinking about making the turf near the user have temporary density when the singularity spawns so that they both can't run and the singularity is grief-proof.

**Comment by Jordie0608** [comment]
I think if people are dumb enough to walk into a singularity they deserve it.
Plus giving turfs density when they normally don't, even for a few seconds, could cause all sorts of shenanigans.

**Pull request merged**